### PR TITLE
Makefile fixes

### PR DESCRIPTION
**Pull request by codido**
The following PR fixes static builds on clean environments, and speeds up successive build times by saving the packages that are dependencies of runc:
```
$ time make
ln -sfn /tmp/runc /tmp/runc/Godeps/_workspace/src/github.com/opencontainers/runc
go build -i -ldflags "-X main.gitCommit=21e3b2f2ea9f08b901e8c20438a74a55287feae7" -tags "seccomp" -o runc .

real	0m2.641s
user	0m8.717s
sys	0m0.742s
$ time make
go build -i -ldflags "-X main.gitCommit=21e3b2f2ea9f08b901e8c20438a74a55287feae7" -tags "seccomp" -o runc .

real	0m1.214s
user	0m1.645s
sys	0m0.159s
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,18 +7,18 @@ RUNC_BUILD_PATH=/go/src/github.com/opencontainers/runc/runc
...
 
 .PHONY=dbuild
 
-all:
-ifneq ($(RUNC_LINK), $(wildcard $(RUNC_LINK)))
-	ln -sfn $(CURDIR) $(RUNC_LINK)
-endif
-	go build -ldflags "-X main.gitCommit=${COMMIT}" -tags "$(BUILDTAGS)" -o runc .
+all: $(RUNC_LINK)
+	go build -i -ldflags "-X main.gitCommit=${COMMIT}" -tags "$(BUILDTAGS)" -o runc .
```
Comment (hqhq) [inline Makefile]: `-i` seems useful, but I don't see it widely used, is it safe to use when depending on packages that will change?

Comment (codido) [inline Makefile]: It seems so, yes:
```
$ time make
go build -i -ldflags "-X main.gitCommit=80cf52436a3f2d85b0e74b4c4045e9cf6fa0b25c" -tags "seccomp" -o runc .

real	0m2.648s
user	0m8.667s
sys	0m0.771s
[runc]$ ls -la Godeps/_workspace/pkg/linux_amd64/github.com/opencontainers/runc/libcontainer
total 1548
drwxrwxr-x. 4 ido ido   4096 Apr 13 08:30 .
drwxrwxr-x. 3 ido ido   4096 Apr 13 08:30 ..
-rw-r--r--. 1 ido ido   3466 Apr 13 08:30 apparmor.a
drwxrwxr-x. 2 ido ido   4096 Apr 13 08:30 cgroups
-rw-r--r--. 1 ido ido 194402 Apr 13 08:30 cgroups.a
drwxrwxr-x. 2 ido ido   4096 Apr 13 08:30 configs
-rw-r--r--. 1 ido ido 323352 Apr 13 08:30 configs.a
-rw-r--r--. 1 ido ido 325120 Apr 13 08:30 criurpc.a
-rw-r--r--. 1 ido ido  14674 Apr 13 08:30 keys.a
-rw-r--r--. 1 ido ido  10976 Apr 13 08:30 label.a
-rw-r--r--. 1 ido ido  64024 Apr 13 08:30 nsenter.a
-rw-r--r--. 1 ido ido 105358 Apr 13 08:30 seccomp.a
-rw-r--r--. 1 ido ido 242788 Apr 13 08:30 specconv.a
-rw-r--r--. 1 ido ido  12738 Apr 13 08:30 stacktrace.a
-rw-r--r--. 1 ido ido  87030 Apr 13 08:30 system.a
-rw-r--r--. 1 ido ido 142792 Apr 13 08:30 user.a
-rw-r--r--. 1 ido ido  16984 Apr 13 08:30 utils.a
[runc]$ touch Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/nsenter/nsexec.c
[runc]$ touch Godeps/_workspace/src/github.com/opencontainers/runc/libcontainer/user/user.go 
[runc]$ time make
go build -i -ldflags "-X main.gitCommit=80cf52436a3f2d85b0e74b4c4045e9cf6fa0b25c" -tags "seccomp" -o runc .

real	0m1.659s
user	0m2.583s
sys	0m0.257s
[runc]$ ls -la Godeps/_workspace/pkg/linux_amd64/github.com/opencontainers/runc/libcontainer
total 1548
drwxrwxr-x. 4 ido ido   4096 Apr 13 08:31 .
drwxrwxr-x. 3 ido ido   4096 Apr 13 08:31 ..
-rw-r--r--. 1 ido ido   3466 Apr 13 08:30 apparmor.a
drwxrwxr-x. 2 ido ido   4096 Apr 13 08:30 cgroups
-rw-r--r--. 1 ido ido 194402 Apr 13 08:30 cgroups.a
drwxrwxr-x. 2 ido ido   4096 Apr 13 08:30 configs
-rw-r--r--. 1 ido ido 323352 Apr 13 08:30 configs.a
-rw-r--r--. 1 ido ido 325120 Apr 13 08:30 criurpc.a
-rw-r--r--. 1 ido ido  14674 Apr 13 08:30 keys.a
-rw-r--r--. 1 ido ido  10976 Apr 13 08:30 label.a
-rw-r--r--. 1 ido ido  64024 Apr 13 08:31 nsenter.a
-rw-r--r--. 1 ido ido 105358 Apr 13 08:30 seccomp.a
-rw-r--r--. 1 ido ido 242788 Apr 13 08:30 specconv.a
-rw-r--r--. 1 ido ido  12738 Apr 13 08:30 stacktrace.a
-rw-r--r--. 1 ido ido  87030 Apr 13 08:30 system.a
-rw-r--r--. 1 ido ido 142792 Apr 13 08:31 user.a
-rw-r--r--. 1 ido ido  16984 Apr 13 08:30 utils.a
[runc]$ time make
go build -i -ldflags "-X main.gitCommit=80cf52436a3f2d85b0e74b4c4045e9cf6fa0b25c" -tags "seccomp" -o runc .

real	0m1.213s
user	0m1.635s
sys	0m0.171s
```

Comment (hqhq) [inline Makefile]: Thanks for clarifying.

**Comment by hqhq** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM
@LK4D4 you want to take a look as well?

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**